### PR TITLE
Add Multiple Digital Notch filter

### PR DESCRIPTION
**Pull request by psomhorst**
Add an MDN filter. 

This PR lacks tests as of yet.

### CODE REVIEW

File: eitprocessing/filters/mdn.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            )
+            warnings.warn(msg, UserWarning, stacklevel=2)
+
+        if self.respiratory_rate > self.heart_rate:
+            msg = (
+                f"The respiratory rate ({self.respiratory_rate:.1f} Hz) is higher than the heart "
+                f"rate ({self.heart_rate:.1f} Hz)."
+            )
+            raise ValueError(msg)
+
```
Comment (JulietteFrancovich) [inline eitprocessing/filters/mdn.py:81]: maybe add check that respiratory and heart rate must be positive? Probably not necessary but if someone puts in a negative heart rate no filters will be applied.

File: eitprocessing/filters/mdn.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                number of harmonics and the frequency bands used for filtering.
+            **kwargs: Additional keyword arguments to pass to the ContinuousData or EITData object (e.g., `label`).
+        """
+        data: np.ndarray
+
+        sample_frequency_, axis_, data = self._validate_arguments(
+            input_data=input_data, sample_frequency=sample_frequency, axis=axis
+        )
+
+        # Ensure the data is filtered up to the point where lower_limit would be larger than the noise frequency limit
```
Comment (JulietteFrancovich) [inline eitprocessing/filters/mdn.py:122]: Why filter up to the point where the lower_limit is larger than the noise frequency limit? It seems a bit counterintuitive to me, since the notch band could extend significantly above the noise limit, potentially filtering frequencies beyond the intended range. I understand the desire to avoid missing harmonics near the noise frequency limit, but perhaps using the notch center frequency (rather than the lower_limit) as the threshold would be more intuitive and ensure the filtered range aligns more closely with the noise limit.

File: tests/test_mdn_filter.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        heart_rate=80 / MINUTE,
+    )
+    with pytest.raises(ValueError, match="Sample frequency must be provided."):
+        mdn_filter.apply(signal)
+
+    with pytest.raises(ValueError, match="Sample frequency must be provided."):
+        mdn_filter.apply(signal, sample_frequency=None)
+
+
+def test_close_respiratory_and_heart_rate(signal_factory: Callable):
```
Comment (JulietteFrancovich) [inline tests/test_mdn_filter.py:185]: Would it make sense to also test the edge case that the respiratory and heart rate are exactly the same?

File: tests/test_mdn_filter.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        heart_rate=80 / MINUTE,
+    )
+
+    filtered_data = mdn_filter.apply(eit_data)
+    filtered_signal = mdn_filter.apply(eit_data.pixel_impedance, sample_frequency=eit_data.sample_frequency, axis=0)
+
+    assert np.allclose(filtered_data.pixel_impedance, filtered_signal)
+
+
+@pytest.mark.parametrize(
```
Comment (JulietteFrancovich) [inline tests/test_mdn_filter.py:123]: Would it make sense to test what happens when sample frequency is very low which could cause aliasing?

File: eitprocessing/filters/mdn.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+from eitprocessing.datahandling.continuousdata import ContinuousData
+from eitprocessing.datahandling.eitdata import EITData
+from eitprocessing.filters import TimeDomainFilter
+
+MINUTE = 60
+NOISE_FREQUENCY_LIMIT: float = 220 / MINUTE
+DEFAULT_AXIS: int = 0
+
+# TODO: centralize settings (these should be shared with e.g. RateDetection)
+UPPER_RESPIRATORY_RATE_LIMIT: float = 85 / MINUTE
```
[the remark this answers is not in the document]
Comment (psomhorst) [reply eitprocessing/filters/mdn.py:21]: These serve more as a check-value to make sure people used Hz instead of bpm. The filters will function fine with higher values (except that the results are probably not great), and they do not have the same limitations as the rate detection algorithm (where the limits are very useful to get the best results). Since I don't know whether the exact same values make the most sense, I didn't immediately import or give the same name to the values.

I would suggest keeping them like this until we have some centralized setting system.

File: eitprocessing/filters/mdn.py (hunk 2 above)
Comment (psomhorst) [reply eitprocessing/filters/mdn.py:125]: All data above the noise limit is filtered using a low pass filter anyway. This implementation only has the risk of filtering the a small section twice, which does not really have implications. I would rather have that than not filtering the upper band.

If the noise limit 220 bpm, the heart rate is 72 bpm, the 3rd harmonic is 216 bpm, then 206-226 bpm **should be** filtered, and 220-226 is filtered twice. 

Note that the description is correct, but the implementation wasn't (it would also filter around the 4th harmonic, 288 bpm). That issue is fixed.

File: tests/test_mdn_filter.py (hunk 3 above)
Comment (psomhorst) [reply tests/test_mdn_filter.py:185]: I added an error (and test) when the RR and HR are equal. However, this does not cover cases where they only differ very slightly. 

We probably should test what the minimum difference should be to have good separation. Probably more than half the notch distance, because at half the notch distances, the second harmonic of the RR lies the notch distance away from the second harmonic of the HR. I don't really think it's worth testing this before merging this PR, because it's quite an edge case.
